homeassistant/components/simplisafe/__init__.py

```diff
@@ -387,23 +382,13 @@ async def async_update_system(system):
 
         for result in results:
             if isinstance(result, InvalidCredentialsError):
-                if self._emergency_refresh_token_used:
-                    raise ConfigEntryAuthFailed(
-                        "Update failed with stored refresh token"
-                    )
-
-                LOGGER.warning("SimpliSafe cloud error; trying stored refresh token")
-                self._emergency_refresh_token_used = True
-
                 try:
-                    await self._api.refresh_access_token(
-                        self.config_entry.data[CONF_TOKEN]
-                    )
+                    self._api = await self._async_get_api()
                     return
                 except SimplipyError as err:
```

Comment:
Above we catch `InvalidCredentialsError` when we raise `ConfigEntryAuthFailed`. Here we catch `SimplipyError`. What's the difference?

Reply:
@MartinHjelmare `SimplypiError` is more generic and could arise for other reasons – like the web service being down. In that case, the user can't actively re-auth anything… But I'm open to suggestions!￼

Reply:
Maybe raise `UpdateFailed` on `SimplypiError` and only on `InvalidCredentialsError` raise `ConfigEntryAuthFailed`?